system_tests/batch_poster_test.go

```diff
@@ -80,24 +79,13 @@ func externalSignerTestCfg(addr common.Address) (*dataposter.ExternalSignerCfg,
 func testBatchPosterParallel(t *testing.T, useRedis bool) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	httpSrv, srv := externalsignertest.NewServer(t)
-	cp, err := externalsignertest.CertPaths()
-	if err != nil {
-		t.Fatalf("Error getting cert paths: %v", err)
-	}
-	t.Cleanup(func() {
-		if err := httpSrv.Shutdown(ctx); err != nil {
-			t.Fatalf("Error shutting down http server: %v", err)
-		}
-	})
```

Comment:
I think you forgot to add the shutdown logic to NewServer function.

Reply:
It's actually already in the NewServer function. I think this code was already redundant.